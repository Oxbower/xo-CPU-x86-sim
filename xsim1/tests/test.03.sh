#!/bin/sh

$1 190 tests/test.xo
