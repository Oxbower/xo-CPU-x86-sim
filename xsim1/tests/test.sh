#!/bin/sh

if ! [ -f tests/test.xo ]; then
  ./xas tests/test.xas tests/test.xo > /dev/null
fi 

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 tests/test.$1.sh ./$2/$3 > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
