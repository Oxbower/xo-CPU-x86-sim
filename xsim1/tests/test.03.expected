PC: 0002, State: 0002
Registers: 
      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0000
Next instruction: loadi 1036, r15

PC: 0006, State: 0002
Registers: 
      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:040c
Next instruction: call 1036

PC: 040c, State: 0002
Registers: 
      r00:0000 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:040a
Next instruction: loadi 1064, r0

PC: 0410, State: 0002
Registers: 
      r00:0428 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:040a
Next instruction: push r0

PC: 0412, State: 0002
Registers: 
      r00:0428 r01:0000 r02:0000 r03:0000 r04:0000 r05:0000 r06:0000 r07:0000
      r08:0000 r09:0000 r10:0000 r11:0000 r12:0000 r13:0000 r14:0000 r15:0408
Next instruction: cld 

Hello world!
Hello world!
